### PR TITLE
Feature/yc 785

### PR DESCRIPTION
**Pull request by nicolas-heigvd**
- [ ] Use DRYer approach
- [x] Fix js bug: TypeError: monetaryAmount is null on hourly rate item creation
- [ ] Translate time sum
- [x] Fix handling of nullable Money field
- [x] make fixed price optional in admin
- [ ] Write fixtures
- [ ] Write test

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -19,13 +22,20 @@ services:
...
     depends_on:
-      - "postgres"
+      postgres:
+        condition: service_healthy
     ports:
       - "${DJANGO_DOCKER_PORT}:9000"
 
   postgres:
-    image: postgis/postgis:13-3.2
+    image: postgis/postgis:16-3.4
```
Comment (monodo) [inline docker-compose.dev.yml:31]: ⚠️ This will, in terme, required from production instances to upgrade postgres servers ⚠️ First check that is is fine for instances admins. That said, I like the idea.

File: geocity/apps/submissions/permissions.py [hunk 2]
```diff
@@ -36,6 +36,33 @@ def has_permission_to_amend_submission(user, submission):
...
+
+
+def has_permission_to_update_service_fees(user, submission):
+    return (
+        user.has_perm("submissions.update_service_fee")
+        and submission.administrative_entity
+        in AdministrativeEntity.objects.associated_to_user(user)
+        and is_backoffice_of_submission(user, submission)
+    )
+
```
Comment (monodo) [inline geocity/apps/submissions/permissions.py:55]: Please find a DRYer solution

File: geocity/apps/submissions/permissions.py [hunk 3]
```diff
@@ -36,6 +36,33 @@ def has_permission_to_amend_submission(user, submission):
...
+
+def has_permission_to_delete_service_fees(user, submission):
+    return (
+        user.has_perm("submissions.delete_service_fee")
+        and submission.administrative_entity
+        in AdministrativeEntity.objects.associated_to_user(user)
+        and is_backoffice_of_submission(user, submission)
+    )
+
```
Comment (monodo) [inline geocity/apps/submissions/permissions.py:64]: Idem

File: geocity/apps/submissions/shortcuts.py [hunk 4]
```diff
@@ -2,20 +2,33 @@
...
     """
-    Return the submission with `submission` or raise an Http404 if there is no such submission. The
+    Return the submission with `submission_id` or raise an Http404 if there is no such submission. The
     submission must either belong to the given user, or the given user should be in the same administrative entity.
     If `statuses` is set and a submission is found but its status doesn't match any value in `statuses`,
     `BadSubmissionStatus` will be raised.
     """
     permit_request = get_object_or_404(
-        models.Submission.objects.filter_for_user(user), pk=submission
+        models.Submission.objects.filter_for_user(user), pk=submission_id
```
Comment (monodo) [inline geocity/apps/submissions/shortcuts.py:16]: What is the rationale for this change ? This function beeing used in many places, are you sure it does't brake things ?

File: geocity/apps/submissions/views.py [hunk 5]
```diff
@@ -137,6 +137,27 @@ def get_submission_for_prolongation(user, submission_id):
...
 
+def get_submission_service_fee(user, submission_id):
+    allowed_statuses = models.Submission.SERVICE_FEES_STATUSES
+    submission = get_submission_for_user_or_404(
+        user,
+        submission_id,
+        statuses=allowed_statuses,
+    )
+
+    return submission
```
Comment (monodo) [inline geocity/apps/submissions/views.py:148]: Move to model instead ?

File: geocity/apps/submissions/views.py [hunk 6]
```diff
@@ -137,6 +137,27 @@ def get_submission_for_prolongation(user, submission_id):
...
+
+
+def get_service_fee(user, service_fee_id):
+    allowed_statuses = models.Submission.SERVICE_FEES_STATUSES
+    service_fee = get_service_fee_for_user_or_404(
+        user,
+        service_fee_id,
+    )
+
+    return service_fee
```
Comment (monodo) [inline geocity/apps/submissions/views.py:158]: idem

File: geocity/apps/submissions/views.py [hunk 7]
```diff
@@ -544,6 +592,26 @@ def handle_form_submission(self, form, action):
             return self.handle_complementary_documents_form_submission(form)
         elif action == models.ACTION_REQUEST_INQUIRY:
             return self.handle_request_inquiry_form_submission(form)
+        elif action == models.ACTION_CREATE_SERVICE_FEE:
+            return self.handle_service_fee_form_submission(form)
+        elif action == models.ACTION_UPDATE_SERVICE_FEE:
+            return self.handle_service_fee_form_submission(form)
+        elif action == models.ACTION_DELETE_SERVICE_FEE:
+            return self.handle_service_fee_form_submission(form)
```
Comment (monodo) [inline geocity/apps/submissions/views.py:600]: Do you really need to discriminate among CRUD action here ?

File: geocity/apps/submissions/views.py [hunk 8]
```diff
@@ -2147,13 +2215,213 @@ def administrative_entities_geojson(request, administrative_entity_id):
...
+        request.user,
+        submission_id,
+    )
+    logger.debug(f"submission admin entites: {submission.administrative_entity}")
+    departments = get_departments(request.user)
+    department = departments.first()
+    # dpts = PermitDepartment.objects.filter(
+    #     administrative_entity=submission.administrative_entity,
+    # )
+    # print(f"dpts: {(dpts)}")
```
Comment (monodo) [inline geocity/apps/submissions/views.py:2234]: remove commented code

**Review by monodo — CHANGES_REQUESTED** [review]
Here is a (very) quick review. Some general considerations:
- In this project, the texts to be translated are in french. I don't not recommend to change this only in this particular PR. Should there be a reason of changing this, it should be done everywhere in a separate PR
- Changing postgres version for dev, might cause bad suprises in prod as tests won't run anymore on the same version
- Looks like the PR is ready for...writing tests

Other comments are in the files. Thanks!

**Comment by monodo** [comment]
- [ ] Add permission for integrator to view/edit service fee model
- [ ] Ensure integrator only sees service fees created by its integrator group

File: docker-compose.dev.yml (hunk 1 above)
Comment (nicolas-heigvd) [reply docker-compose.dev.yml]: Restored back to main.

File: geocity/apps/submissions/shortcuts.py (hunk 4 above)
Comment (nicolas-heigvd) [reply geocity/apps/submissions/shortcuts.py:16]: This is only a cosmetic change.

File: geocity/apps/submissions/views.py (hunk 7 above)
Comment (nicolas-heigvd) [reply geocity/apps/submissions/views.py]: Outdated.

File: geocity/apps/submissions/views.py (hunk 6 above)
Comment (nicolas-heigvd) [reply geocity/apps/submissions/views.py]: Outdated.

File: geocity/apps/submissions/views.py (hunk 8 above)
Comment (nicolas-heigvd) [reply geocity/apps/submissions/views.py]: Done.

File: geocity/apps/submissions/views.py (hunk 5 above)
Comment (nicolas-heigvd) [reply geocity/apps/submissions/views.py:150]: Not crictical for the moment. The feature is subject to near future changes.

File: geocity/apps/submissions/permissions.py (hunk 3 above)
Comment (nicolas-sitylb) [reply geocity/apps/submissions/permissions.py:46]: Outdated.

File: geocity/apps/submissions/payments/models.py [hunk 9]
```diff
@@ -116,3 +119,186 @@ def get_refund_pdf(self, read=False):
...
+    updated_by = models.ForeignKey(
+        User,
+        null=True,
+        on_delete=models.SET_NULL,
+        max_length=255,
+        verbose_name=_("Mise à jour par"),
+        related_name="service_fee_updated_by",
+        help_text=_("La prestation a été mise à jour par cet utilisateur."),
+    )
+    permit_department = models.ForeignKey(
```
Comment (monodo) [inline geocity/apps/submissions/payments/models.py:219]: Can you not find the departement from the user groups and the submission entity ?
